hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/bucket/TestBucketCache.java

```diff
@@ -226,10 +226,19 @@ public void testHeapSizeChanges() throws Exception {
 
   public static void waitUntilFlushedToBucket(BucketCache cache, BlockCacheKey cacheKey)
     throws InterruptedException {
-    while (!cache.backingMap.containsKey(cacheKey) || cache.ramCache.containsKey(cacheKey)) {
-      Thread.sleep(100);
+    long timeout = 120000;
+    try {
+      while (!cache.backingMap.containsKey(cacheKey) || cache.ramCache.containsKey(cacheKey)) {
+        Thread.sleep(100);
+        if (timeout <= 0) {
+          break;
+        }
+        timeout -= 100;
+      }
+    } finally {
+      Thread.sleep(1000);
```

Comment:
Hi @wchevreuil , the config object is not available with us here. Probably I can leave out the change this. I had tried to change it to avoid infinite wait.

Reply:
Just create one `HBaseConfiguration.create()` and pass along? Waiter only uses it to check for the `hbase.test.wait.for.ratio` property anyways.

Reply:
ack